src/legacy/ui/public/saved_objects/components/saved_object_finder.tsx

```diff
@@ -248,73 +328,169 @@ class SavedObjectFinder extends React.Component<SavedObjectFinderProps, SavedObj
+                  defaultMessage="Sort: "
+                />
+                <EuiIcon type={this.getCurrentSortIcon()} />
+              </EuiButtonEmpty>
+            </EuiToolTip>
+            {this.props.showFilter && availableSavedObjectMetaData.length > 1 && (
+              <EuiPopover
+                id="popover"
+                panelClassName="euiFilterGroup__popoverPanel"
+                panelPaddingSize="none"
+                isOpen={this.state.filterOpen}
+                closePopover={() => this.setState({ filterOpen: false })}
+                button={
+                  <EuiButtonEmpty
+                    onClick={() =>
+                      this.setState(({ filterOpen }) => ({
+                        filterOpen: !filterOpen,
+                      }))
+                    }
+                    iconSide="right"
+                    iconType="filter"
+                    size="s"
+                    color="text"
+                    data-test-subj="savedObjectFinderFilterButton"
+                  >
+                    <FormattedMessage
+                      id="common.ui.savedObjects.finder.filterButtonLabel"
+                      defaultMessage="Filter"
+                    />
+                  </EuiButtonEmpty>
+                }
+              >
+                <div className="euiFilterSelect__items">
+                  <EuiText className="euiFilterSelectItem savedObjectFilterHeading" size="xs">
+                    <FormattedMessage
+                      id="common.ui.savedObjects.finder.filterLabel"
+                      defaultMessage="Only show the following objects:"
+                    />
+                  </EuiText>
+                  {availableSavedObjectMetaData.map(metaData => (
+                    <EuiFilterSelectItem
+                      key={metaData.type}
+                      checked={this.state.filteredTypes.includes(metaData.type) ? 'on' : undefined}
+                      data-test-subj={`savedObjectFinderFilter-${metaData.type}`}
+                      onClick={() => {
+                        this.setState(({ filteredTypes }) => ({
+                          filteredTypes: filteredTypes.includes(metaData.type)
+                            ? filteredTypes.filter(t => t !== metaData.type)
+                            : [...filteredTypes, metaData.type],
+                          page: 0,
+                        }));
+                      }}
+                    >
+                      {metaData.name}
+                    </EuiFilterSelectItem>
+                  ))}
+                </div>
+              </EuiPopover>
+            )}
+          </EuiFilterGroup>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>{this.props.callToActionButton}</EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
 
-        {this.props.callToActionButton && (
-          <EuiFlexItem grow={false}>{this.props.callToActionButton}</EuiFlexItem>
-        )}
+  private renderLoadingIndicator() {
```

Comment:
The loading indicator replaces the list every time the user types another character into the search field. It's quite jarring to see that much content swapping. Is it possible to not show "loading" unless it's the initial loading of all objects?

<img src="https://d.pr/free/i/ITsVVZ+" width="50%" />

Reply:
Makes sense, I also added an EuiProgress underneath the search bar because it feels a bit strange not knowing whether it searches at all.